projects/igniteui-angular/src/lib/grids/api.service.ts

```diff
@@ -557,20 +544,20 @@ export class GridBaseAPIService<T extends GridType> implements GridServiceType {
      * @param rowCurrentValue Current value of the row as it is with applied previous transactions
      * @param rowNewValue New value of the row
      */
-    protected updateData(grid, rowID, rowValueInDataSource: any, rowCurrentValue: any, rowNewValue: { [x: string]: any }, validity?) {
-        
+    protected updateData(grid, rowID, rowValueInDataSource: any, rowCurrentValue: any, rowNewValue: { [x: string]: any }) {
         const transaction: Transaction = {
             id: rowID,
             type: TransactionType.UPDATE,
-            newValue: rowNewValue,
-            validity: validity
+            newValue: rowNewValue
         };
         if (grid.transactions.enabled) {
             grid.transactions.add(transaction, rowCurrentValue);
         } else {
             mergeObjects(rowValueInDataSource, rowNewValue);
         }
-        grid.transactions.addValidation(transaction, rowCurrentValue);
+        const formGroup = this.grid.validation.getFormGroup(rowID);
+        const validation = grid.validation as IgxGridValidationService;
+        validation.add(rowID, formGroup)
```

Comment:
What we might want to do though (after checking with the CRUD service to avoid in-edit double dips, since UI edits end up here too), is to check if value is updated for a validated record/cell and update it's status accordingly.